File: VonexDealer.Vue/ClientApp/models/Company.js
export class Company {
    constructor({ companyID = 0,
        companyName = null,
        tradingName = null,
        entityType = null,
        directorSurname = null,
        abn = null,
        acn = null,
        driversLicenseNumber = null,
        dateOfBirth = new Date("1901-01-01"),
        directorGivenName = null,
        otherOrganisation = false,
        guaranteeAttached = false,
        dobConsumer = new Date("1901-01-01"),
        isdirectorSurnameRequired = false,
        isdateOfBirthRequired = false
        } = {}) {
        this.companyID = companyID;
        this.companyName = companyName;
        this.tradingName = tradingName;
        this.entityType = entityType;
        this.directorSurname = directorSurname;
        this.abn = abn;
        this.acn = acn;
        this.driversLicenseNumber = driversLicenseNumber;
        this.dateOfBirth = dateOfBirth;
        this.directorGivenName = directorGivenName;
        this.otherOrganisation = otherOrganisation;
        this.guaranteeAttached = guaranteeAttached;
        this.dobConsumer = dobConsumer;
        this.isdirectorSurnameRequired = isdirectorSurnameRequired;
        this.isdateOfBirthRequired = isdateOfBirthRequired;
    }
}

export function createCompany(data) {
    return Object.freeze(new Company(data));
}

File: VonexDealer.API/Models/Order/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace VonexDealer.API.Models.Order
{
    public partial class Company
    {
        public Int64 CompanyID { get; set; }
        public string CompanyName { get; set; }
        public string TradingName { get; set; }
        public string EntityType { get; set; }
        public string DirectorGivenName { get; set; }
        public string DirectorSurname { get; set; }
        public string ABN { get; set; }
        public string ACN { get; set; }
        public string DriversLicenseNumber { get; set; }
        public DateTime DateOfBirth { get; set; }
        [NotMapped]
        public DateTime DobConsumer { get; set; }
        [NotMapped]
        public bool guaranteeAttached { get; set; }
        
        public bool isdateOfBirthRequired { get; set; }
       
        public bool isdirectorSurnameRequried { get; set; }
        [NotMapped]
        public bool otherOrganisation { get; set; }
        //public Int64? CustomerID { get; set; }
    }
}
